python/ray/function_manager.py

```diff
@@ -87,9 +89,9 @@ def from_bytes_list(cls, function_descriptor_list):
             return FunctionDescriptor.for_driver_task()
         elif (len(function_descriptor_list) == 3
               or len(function_descriptor_list) == 4):
-            module_name = function_descriptor_list[0].decode()
-            class_name = function_descriptor_list[1].decode()
-            function_name = function_descriptor_list[2].decode()
+            module_name = six.ensure_str(function_descriptor_list[0])
```

Comment:
does `six.ensure_str` do the same thing as `ray.utils.decode`?

Reply:
@robertnishihara In Python3, yes, it is. However, in Python 2, `ray.utils.decode` will return the type of `u""`. Maybe in the future when Python 2 is no longer supported, we can change `six.ensure_str` back to `ray.utils.decode`

Reply:
Generally, I think we should change all `encode/decode` to `six.ensure_binary/ensure_str`. It's easier to maintain compatibility between python 2 and 3.

Reply:
I just checked and `ray.utils.decode` does not return type `u""`, so I think they are identical in Python 2 and Python 3. I'm ok with using `six.ensure_str` instead of `ray.utils.decode`, but then please do it everywhere for consistency.

Reply:
Here is the example in Python2:
```python
Python 2.7.5 (default, Jun 13 2018, 16:41:29)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a="abcd"
>>> a.decode("ascii")
u'abcd'
```

Reply:
@guoyuhong , @robertnishihara was referring to `ray.utils.decode`, which should be the same as `six.ensure_str`. 
I'll do the replacement in a different PR. Because there're many occurrences.

Reply:
How about using `six.ensure_str` directly in `ray.utils.decode`. In that case, we don't need to change anything.

Reply:
@raulchen I just got your point. There are many `decode` calls needs to be changed. That is fine.

Reply:
That's fine as well.